### PR TITLE
created google login and did bug fixes

### PR DESCRIPTION
**Pull request by ayushkumar2808**

### CODE REVIEW

File: client/src/components/pages/GoogleAuth.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { useNavigate } from "react-router-dom";
+const GoogleAuth = () => {
+  const { state, dispatch } = useContext(UserContext);
+  const navigate = useNavigate();
+  useEffect(() => {
+    /* global google */
+
+    google.accounts.id.initialize({
+      client_id:
+        "382413014720-hc9v4e6gnh52giikp9u6f0qa79qj0f11.apps.googleusercontent.com",
```
Comment (PatelNeelMahesh) [inline client/src/components/pages/GoogleAuth.js]: You might want to add client_id in .env file.

File: server/routes/auth.js [hunk 2]
```diff
@@ -2,13 +2,16 @@ const express = require('express')
...
-
+const mongoose  = require('mongoose')
 const User = require('../models/userModel')
 const bcrypt = require('bcrypt')
 const jwt  = require('jsonwebtoken')
 const secret = process.env.SECRET
 
 
+const { OAuth2Client } = require("google-auth-library");
+const client = new OAuth2Client("382413014720-hc9v4e6gnh52giikp9u6f0qa79qj0f11.apps.googleusercontent.com");
```
Comment (PatelNeelMahesh) [inline server/routes/auth.js]: Get variable from .env

**Review by PatelNeelMahesh — CHANGES_REQUESTED** [review]
After doing required changes you can merge pull request.

File: client/src/components/pages/GoogleAuth.js (hunk 1 above)
Comment (ayushkumar2808) [reply client/src/components/pages/GoogleAuth.js]: ok, I will add it.

**Pull request merged**